pkg/codegen/pcl/resource.go

```diff
@@ -89,11 +94,21 @@ func (r *Resource) Traverse(traverser hcl.Traverser) (model.Traversable, hcl.Dia
 	return r.VariableType.Traverse(traverser)
 }
 
-// Name returns the name of the resource.
+// Deprecated: Name returns the variable or declaration name of the resource.
 func (r *Resource) Name() string {
 	return r.Definition.Labels[0]
 }
```

Comment:
Are we depreciating this now? We still use it in our code. See `+python/gen_program.go:358`. What happened to `LexicalName`?

I don't think we should depreciate this, since it's an internal API. I think we should change the name to `LexicalName`.

Reply:
I think `Name()` since it's exported might be used by tfgen et al. This was a snafu while I was in the process of adding `LexicalName()`, and I temporarily had two methods, a deprecated `Name()` and a new `LexicalName()`.

The refactor to rename Name to LexicalName seems out of scope. I think the comments are fine, _for now_, but we should revisit in a few weeks. I'll create an issue and assign it to myself.

Reply:
I've force-pushed a commit removing this.

Reply:
- https://github.com/pulumi/pulumi/issues/9469